### PR TITLE
Fix #206 : Add tests for Assigning/Revoking admin DAO

### LINKED ISSUE #206: Add more tests to AdminDAO
## Description
As a developer,
I need have more test cases tested,
so that I can cover most of the logic.

### Update [Required]
- [ ] Create more tests that test the AdminDao logic

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by sumit-badsara**
##### Added the following tests to existing admin tests: in `tests/admin/test_dao.py`
- Assigning a user which is already an admin -> should receive a message saying that the user is already an admin
- Assigning a user as an admin, when the current user is not an admin -> should receive an error message
- Assigning a non-existent user -> should receive error message
- Revoking a user as an admin, when the current user is not an admin -> should receive an error message

#### Added error messages in `app/api/dao/admin.py`
- Added Error Message if a non admin user tries to assign someone as new admin.
- Added Error Message if a non admin user tries to revoke admin status of someone.

Fixes #206

### Type of Change:
**Delete irrelevant options.**

- Code
- Quality Assurance
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
`python -m unittest discover tests`

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/admin/test_dao.py
Comment (isabelcosta) [inline tests/admin/test_dao.py]:
```suggestion
    Checks whether a new admin can be assigned by existing admin.
```

File: tests/admin/test_dao.py
Comment (isabelcosta) [inline tests/admin/test_dao.py]:
```suggestion
    Checks whether a new admin can be assigned by normal user.
```

File: tests/admin/test_dao.py
Comment (isabelcosta) [inline tests/admin/test_dao.py]:
```suggestion
    Checks if a user tries to self-assign admin role. 
```

File: tests/admin/test_dao.py
Comment (isabelcosta) [inline tests/admin/test_dao.py]:
```suggestion
    Checks whether a user is trying to revoke other user's admin privileges. 
```

File: app/api/dao/admin.py
Comment (isabelcosta) [inline app/api/dao/admin.py]:
```suggestion
                return messages.USER_REVOKE_NOT_ADMIN, 403
```

the message being used should be about revoking a user and not assigning

File: app/api/dao/admin.py
Comment (isabelcosta) [inline app/api/dao/admin.py]: You can remove this comment, because this is not only for tests, but actually "production" code that was missing ;)

File: app/api/dao/admin.py
Comment (isabelcosta) [inline app/api/dao/admin.py]: You can remove this comment, because this is not only for tests, but actually "production" code that was missing ;)

File: tests/admin/test_dao.py
Comment (isabelcosta) [inline tests/admin/test_dao.py]:
```suggestion
        self.assertEqual((messages.USER_REVOKE_NOT_ADMIN, 403), dao_result)
```
Once you fix the message being returned for revoking an admin, you'll have to fix this here

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@BadduCoder great work with this PR :tada: 
I requested changes for some small typos and a logic change.
After you do this, I can merge the PR.

**Comment by isabelcosta** [comment]
@BadduCoder  Usually we prefer that contributors fix one issue per pull request (PR). Also we prefer that you ask to work on an issue, before sending a PR, so we can better manage who works on what, and make sure "first-timers-only" are done by newcomers to the repository. As you are a newcomer its ok, that you're solving a "first-timers-only". I just wanted to let you know of this.

**Comment by isabelcosta** [comment]
I will edit the description to your PR, because this actually does not fix the tests for the API, but actually fix the tests for the AdminDAO. This just means you're fixing another issue @BadduCoder

**Comment by sumit-badsara** [comment]
@isabelcosta Yeah sure! Thanks for the help, i will do the changes .

**Review by isabelcosta — DISMISSED** [review]
Thank you for doing the changes @BadduCoder and for your contribution to our tests 🎉

**Comment by isabelcosta** [comment]
@BadduCoder I will merge this PR once I review some PRs on Mentorship Android (because we still have the problem of database not being persistent, once I merge this all data used by android contributors is gone)

**Comment by sumit-badsara** [comment]
@isabelcosta Sure! Would love to contribute more :grinning:

**Comment by isabelcosta** [comment]
@BadduCoder you should commit the last commit from the new issue you're working on, on a separate branch and pull request, to separate the scope of code review of both issues. Can you please remove this commit from this PR? This made my previous review obsolete :/

**Comment by sumit-badsara** [comment]
Yes @isabelcosta, I have reverted back the commit. Next time i will take care about this for sure. Thanks for helping out!

**Review by isabelcosta — APPROVED** [review]

**Pull request merged**
